src/mds/events/ESession.h

```diff
@@ -30,26 +30,20 @@ class ESession : public LogEvent {
   version_t inotablev{0};
 
   // Client metadata stored during open
-  std::map<std::string, std::string> client_metadata;
+  client_metadata_t client_metadata;
 
  public:
   ESession() : LogEvent(EVENT_SESSION), open(false) { }
   ESession(const entity_inst_t& inst, bool o, version_t v,
-      const std::map<std::string, std::string> &cm) :
+	   client_metadata_t &cm) :
     LogEvent(EVENT_SESSION),
-    client_inst(inst),
-    open(o),
-    cmapv(v),
-    inotablev(0),
-    client_metadata(cm) {
-  }
+    client_inst(inst), open(o), cmapv(v), inotablev(0) {
+      client_metadata.swap(cm);
```

Comment:
This is pretty evil. I would prefer if you made the constructor arg `client_metadata_t &&cm` so that it (should) force the caller to `std::move` the argument. Then it also does an r-value construct of `client_metadata` so that no temporaries are created by the default constructor to `client_metadata`.